### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by donny-dont**
Change the slack client to `slack-go/slack` since `bluele/slack` is no longer being developed. Update to v3 of `gotest.tools`.

The `slack-go/slack` code expects attachments as an instance not a pointer so move around the code to add Text. Add `drone-slack` as the author of the message.

Update expected output of the test to reflect new library.

### CODE REVIEW

**Review by tphoney — APPROVED** [review]

**Comment by tphoney** [comment]
great work @donny-dont

File: plugin.go [hunk 1]
```diff
@@ -95,6 +96,7 @@ func (p Plugin) Exec() error {
 	attachment := slack.Attachment{
 		Color:      p.Config.Color,
 		ImageURL:   p.Config.ImageURL,
+		AuthorName: "drone-slack",
```
Comment (tonglil) [inline plugin.go:99]: Is it possible to omit this? It shows up as kind of crowded in Slack

Comment (donny-dont) [reply plugin.go:99]: Where is it appearing for you? I didn't see anything in our slack instance.

Comment (tonglil) [reply plugin.go:99]: AFAICT it started today (I guess a latest release got published?). 

image

Comment (donny-dont) [reply plugin.go:99]: #152

**Pull request merged**